server/src/main/java/org/elasticsearch/index/mapper/IpPrefixAutomatonUtil.java

```diff
@@ -105,8 +105,13 @@ private static Automaton getIpv6Automaton(String ipPrefix) {
                         ipv6Automaton = concatenate(ipv6Automaton, Operations.repeat(Automata.makeChar(0)));
                     }
                 } else {
+                    // potentially partial block
+                    if (groupsAdded == 0 && ONLY_ZEROS.matcher(group).matches()) {
+                        // we ignore only-zero prefixes for ipv6, otherwise those would also match all ipv4 addresses
+                        return EMPTY_AUTOMATON;
```

Comment:
The reasoning here is: if we expand a user inputing only "0" and allow this to match ipv6 addresses, this would include things like 0000::127.0.0.1 (and all other ipv4 addresses) which then is returned as "127.0.0.1", but this is counterintuitive when starting with a prefix. so "0*" prefixes will only expand to ipv4 addresses like "0.1.2.3".

Reply:
This is very helpful context, and I think it would be useful to expand the code comment to include it.